### PR TITLE
Implementing_min_order

### PR DESCRIPTION
**Pull request by rajlash**

### CODE REVIEW

File: app/models/terminal.rb [hunk 1]
```diff
@@ -22,10 +22,13 @@ class Terminal < ApplicationRecord
...
   before_save :titleize_name
 
-  def ordered_amount(order = nil)
-    (Order.where(status: 'pending', terminal_id: id, date: Time.zone.today)
-          .sum(:total_cost) * (100 + tax.to_f) / 100
-    ).round - (order ? order.total_cost + order.tax : 0)
+  def ordered_amount(order_id = nil)
+    (
+      Order
+      .where(status: 'pending', terminal_id: id, date: Time.zone.today)
```
Comment (kirandeshmukh) [inline app/models/terminal.rb]: What about the orders with the status "placed"? Please consider their impact also.

File: app/models/terminal.rb [hunk 2]
```diff
@@ -22,10 +22,13 @@ class Terminal < ApplicationRecord
   before_validation :remove_space, :active_must_accept_boolean_only
   before_save :titleize_name
 
-  def ordered_amount(order = nil)
-    (Order.where(status: 'pending', terminal_id: id, date: Time.zone.today)
-          .sum(:total_cost) * (100 + tax.to_f) / 100
-    ).round - (order ? order.total_cost + order.tax : 0)
+  def ordered_amount(order_id = nil)
```
Comment (kirandeshmukh) [inline app/models/terminal.rb]: Please add appropriate comments about why we are doing this change

Comment (kirandeshmukh) [inline app/models/terminal.rb]: Please add or update the test case for it.

File: app/models/terminal.rb (hunk 1 above)
Comment (rajlash) [reply app/models/terminal.rb]: Placed orders won't affect the employee's order which is going to be placed or confirmed.

Comment (kirandeshmukh) [reply app/models/terminal.rb]: Ok.

File: app/models/terminal.rb (hunk 2 above)
Comment (rajlash) [reply app/models/terminal.rb]: Test case added.

Comment (rajlash) [reply app/models/terminal.rb]: Comment added for the method.

File: app/models/terminal.rb [hunk 3]
```diff
@@ -22,10 +22,14 @@ class Terminal < ApplicationRecord
   before_validation :remove_space, :active_must_accept_boolean_only
   before_save :titleize_name
 
-  def ordered_amount(order = nil)
-    (Order.where(status: 'pending', terminal_id: id, date: Time.zone.today)
-          .sum(:total_cost) * (100 + tax.to_f) / 100
-    ).round - (order ? order.total_cost + order.tax : 0)
+  #Pending orders' amount from a terminal without considering an order if order_id is passed
```
Comment (kirandeshmukh) [inline app/models/terminal.rb]: This comment seems to be confusing..what is mean by order_id is passed?

File: test/models/terminal_test.rb [hunk 4]
```diff
@@ -193,4 +193,14 @@ class TerminalTest < ActiveSupport::TestCase
     assert_equal 2, @terminal.ordered_amount
   end
 
+  test '#ordered_amount returned by terminal do not consider passed order' do
```
Comment (kirandeshmukh) [inline test/models/terminal_test.rb]: what is mean by passed order? Can u please use the word which will be understood by layman?

File: app/models/terminal.rb (hunk 3 above)
Comment (rajlash) [reply app/models/terminal.rb]: Updated code.

File: test/models/terminal_test.rb (hunk 4 above)
Comment (rajlash) [reply test/models/terminal_test.rb]: Updated code.

**Pull request merged**